collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -125,7 +127,12 @@ public boolean canAddRepeat() {
         }
     }
 
-    public void moveForward() {
+    public void moveForward(FormController formController) {
+        if (this.formController == null) {
+            Timber.w("Null formController");
+            this.formController = formController;
+        }
+
         try {
             formController.stepToNextScreenEvent();
```

Comment:
All the subsequent calls end up using the parameter value. I assume that the singleton form controller is more likely to be the true, authoritative one. It might be possible for the `ViewModel`s' form controllers to be set but not the same as the singleton (though I really hope not).